### PR TITLE
Webpack and VS Code Auth for WebExt

### PR DESCRIPTION
**Pull request by nturinski**
Incorporates the changes to make this run as a web extension. 

Note that if you debug this in VS Code Desktop as a web extension, the `UIKind` _does not_ change to reflect that. It effectively makes testing/debugging impossible 🙄

I had to change the name of the commands of the `extension.ts` to activate/deactivate since that's what VS Code is expecting. VS Code Web doesn't allow more than one file, so you can't use `main.js` as an entry point without breaking a bunch of things.

Also, path and open do not work in VS Code Web so those were replaced.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -577,6 +578,7 @@
         "@vscode/test-electron": "^2.1.5",
         "eslint": "^7.25.0",
         "eslint-plugin-import": "^2.22.1",
+        "fs-extra": "^11.1.0",
```
Comment (bwateratmsft) [inline package.json:581]: Do we actually need `fs-extra` (even as a dev dependency)?

File: src/services/AzureSubscriptionProvider.ts [hunk 2]
```diff
@@ -3,31 +3,17 @@
 *  Licensed under the MIT License. See License.md in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 import * as arm from '@azure/arm-subscriptions';
+import type { Environment } from '@azure/ms-rest-azure-env';
 import { uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureSubscription } from '@microsoft/vscode-azext-utils/hostapi.v2';
 import * as vscode from 'vscode';
+import { AzureLoginStatus } from '../tree/azure-account.api';
```
Comment (bwateratmsft) [inline src/services/AzureSubscriptionProvider.ts:10]: Let's just dump this now...

File: src/services/AzureSubscriptionProvider.ts [hunk 3]
```diff
@@ -3,31 +3,17 @@
 *  Licensed under the MIT License. See License.md in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 import * as arm from '@azure/arm-subscriptions';
+import type { Environment } from '@azure/ms-rest-azure-env';
 import { uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureSubscription } from '@microsoft/vscode-azext-utils/hostapi.v2';
```
Comment (bwateratmsft) [inline src/services/AzureSubscriptionProvider.ts:8]: Can we use the latest that has this in the RG package?

File: src/tree/azure/VSCodeAuthentication.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+export function createSubscriptionContext(subscription: AzureSubscription): ISubscriptionContext {
+    return {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        environment: {} as Environment,
+        isCustomCloud: false,
+        subscriptionDisplayName: subscription.name,
+        subscriptionId: subscription.subscriptionId,
+        subscriptionPath: '',
+        tenantId: '',
+        userId: '',
```
Comment (bwateratmsft) [inline src/tree/azure/VSCodeAuthentication.ts:50]: I'm concerned about potential impact of putting these strings in blank. The eventual consumer of the `ISubscriptionContext` might not expect that.

File: src/utils/openUrl.ts [hunk 5]
```diff
@@ -3,11 +3,9 @@
...
  *--------------------------------------------------------------------------------------------*/
 
-import * as open from 'open';
+import * as vscode from 'vscode';
 
 export async function openUrl(url: string): Promise<void> {
     // Using this functionality is blocked by https://github.com/Microsoft/vscode/issues/85930
-    // await vscode.env.openExternal(vscode.Uri.parse(url));
-
-    await open(url);
```
Comment (bwateratmsft) [inline src/utils/openUrl.ts:12]: Woo get rid of `open`!

File: package.json (hunk 1 above)
Comment (nturinski) [reply package.json:581]: This is just for the gulpfile, so we technically don't need it. But we can't use the vscode `fs`  to replace it.

File: src/tree/azure/VSCodeAuthentication.ts (hunk 4 above)
Comment (nturinski) [reply src/tree/azure/VSCodeAuthentication.ts:50]: Should be resolved Megan's changes

File: api/.npmignore [hunk 6]
```diff
@@ -13,7 +13,10 @@ test-results.xml
 .eslintrc.js
 resources/*.png
 *.tsbuildinfo
+<<<<<<< HEAD
 docs/
+=======
+>>>>>>> 4c48d43 (Create package for consuming extension API (#530))
```
Comment (bwateratmsft) [inline api/.npmignore:19]: Oops

**Pull request merged**